packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m

```diff
@@ -225,7 +225,9 @@ - (NSDictionary *)constantsToExport {
     }
 
     if ([permissions[@"providesAppNotificationSettings"] isEqual:@(YES)]) {
-      options |= UNAuthorizationOptionProvidesAppNotificationSettings;
+      if (@available(iOS 12.0, *)) {
```

Comment:
As a new year's resolution, I'm hoping to expand react-native-firebase utility to all of the platforms supported by the underlying SDKs, as such, this completes the guard here to include the complete list from https://developer.apple.com/documentation/usernotifications/unauthorizationoptions/unauthorizationoptionprovidesappnotificationsettings?language=objc - minus 'iPadOS' which is not a valid `available` platform (yet?)

```suggestion
      if (@available(iOS 12, macOS 10.14, macCatalyst 13, tvOS 12, watchOS 5, *)) {
```

I have tested this and it compiles locally